tests/phpunit/tests/functions/wp_get_nocache_headers.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+/**
+ * @group functions.php
+ * @covers ::wp_get_nocache_headers
+ */
+class Tests_Functions_wp_get_nocache_headers extends WP_UnitTestCase {
+
+	public function test_wp_get_nocache_headers() {
+		$headers = wp_get_nocache_headers();
+
+		$this->assertIsArray( $headers, 'array returned' );
+		$this->assertContains( 'no-cache, must-revalidate, max-age=0', $headers, 'has no cache header' );
+		$this->assertArrayHasKey( 'Expires', $headers, 'has Expires key' );
+		$this->assertArrayHasKey( 'Cache-Control', $headers, 'has Expires key' );
+		$this->assertArrayHasKey( 'Last-Modified', $headers, 'has Last-Modified key' );
+		$this->assertFalse( $headers['Last-Modified'], 'Last-Modified key is false' );
+
+		add_filter(
+			'nocache_headers',
+			function () {
+
+				return array( 'filter_name' => 'nocache_headers' );
+			}
+		);
+
+		$headers = wp_get_nocache_headers();
+		$this->assertEquals( 'nocache_headers', $headers['filter_name'], 'filter was applied' );
+	}
```

Comment:
This test method can be updated to:
- split the tests into separate test methods.
  - One to test `wp_get_nocache_headers()`
  - One to test the `nocache_headers` filter.
- indicate the ticket that added the test methods.
- test the result of `wp_get_nocache_headers()` against an array using `$this->assertSameSetsWithindex()`.
- change the closure to a `static function()`. This is because the closure does not use `$this`.

This results in:
- a clear distinction between what is being tested.
- a reduction in the number of assertions and `$message` parameters.
- stricter testing.
- improved performance.

```suggestion
	/**
	 * @ticket 54490
	 */
	public function test_wp_get_nocache_headers() {
		$this->assertSameSetsWithIndex(
			array(
				'Expires'       => 'Wed, 11 Jan 1984 05:00:00 GMT',
				'Cache-Control' => 'no-cache, must-revalidate, max-age=0',
				'Last-Modified' => false,
			),
			wp_get_nocache_headers()
		);
	}

	/**
	 * @ticket 54490
	 */
	public function test_filter_nocache_headers() {
		add_filter(
			'nocache_headers',
			static function() {
				return array( 'filter_name' => 'nocache_headers' );
			}
		);

		$this->assertSameSetsWithIndex(
			array(
				'filter_name'   => 'nocache_headers',
				'Last-Modified' => false,
			),
			wp_get_nocache_headers()
		);
	}
```